File: language-support/js/daml-ledger/src/mapping/completion_end_response.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as mapping from '.';
import * as ledger from '..';
import * as grpc from 'daml-grpc';

export const CompletionEndResponse: mapping.Mapping<grpc.CompletionEndResponse, ledger.CompletionEndResponse> = {
    toObject(message: grpc.CompletionEndResponse): ledger.CompletionEndResponse {
        return {
            offset: mapping.LedgerOffset.toObject(message.getOffset()!)
        };
    },
    toMessage(response: ledger.CompletionEndResponse): grpc.CompletionEndResponse {
        const result = new grpc.CompletionEndResponse();
        result.setOffset(mapping.LedgerOffset.toMessage(response.offset));
        return result;
    }
}

File: sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/CompletionEndResponse.java
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates.
// Proprietary code. All rights reserved.

package com.daml.ledger.javaapi.data;

import com.daml.ledger.api.v1.CommandCompletionServiceOuterClass;
import java.util.Objects;

public final class CompletionEndResponse {

  private final LedgerOffset offset;

  public CompletionEndResponse(LedgerOffset offset) {
    this.offset = offset;
  }

  public static CompletionEndResponse fromProto(
      CommandCompletionServiceOuterClass.CompletionEndResponse response) {
    return new CompletionEndResponse(LedgerOffset.fromProto(response.getOffset()));
  }

  public LedgerOffset getOffset() {
    return offset;
  }

  @Override
  public String toString() {
    return "CompletionEndResponse{" + "offset=" + offset + '}';
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    CompletionEndResponse that = (CompletionEndResponse) o;
    return Objects.equals(offset, that.offset);
  }

  @Override
  public int hashCode() {

    return Objects.hash(offset);
  }
}
